pandas/core/strings.py

```diff
@@ -1058,7 +1058,8 @@ class StringMethods(object):
     """
 
     def __init__(self, series):
-        self.series = series
+        # can't use direct assignment as we prevent that in __set_item__
+        object.__setattr__(self, "series", series)
```

Comment:
so instead of all of this complication then
make a new mixim in base and just include it where needed